generator/src/test/java/com/scottlogic/deg/generator/cucumber/steps/StringValueStep.java

```diff
@@ -32,12 +32,45 @@ public void producedDataShouldContainStringValuesForField(String fieldName){
         helper.assertFieldContainsNullOrMatching(fieldName, String.class);
     }
 
+    @Then("{fieldVar} contains anything but string data")
+    public void producedDataShouldContainAnythingButStringValuesForField(String fieldName){
+        helper.assertFieldContainsNullOrNotMatching(fieldName, String.class);
+    }
+
     @Then("{fieldVar} contains strings of length between {int} and {int} inclusively")
     public void producedDataShouldContainStringValuesInRangeForField(String fieldName, int minInclusive, int maxInclusive){
         helper.assertFieldContainsNullOrMatching(
             fieldName,
             String.class,
-            value -> isLongerThanOrEqual(value, minInclusive) && isShorterThanOrEqual(value, maxInclusive));
+            isLengthBetweenInclusively(minInclusive, maxInclusive));
+    }
+
+    @Then("{fieldVar} contains strings shorter than or equal to {int}")
+    public void producedDataShouldContainStringValuesShorterThanForField(String fieldName, int shorterThanInclusive){
+        helper.assertFieldContainsNullOrMatching(
+            fieldName,
+            String.class,
+            value -> isShorterThanOrEqual(value, shorterThanInclusive));
+    }
+
+    @Then("{fieldVar} contains strings longer than or equal to {int}")
+    public void producedDataShouldContainStringValuesLongerThanForField(String fieldName, int longerThanInclusive){
+        helper.assertFieldContainsNullOrMatching(
+            fieldName,
+            String.class,
+            value -> isLongerThanOrEqual(value, longerThanInclusive));
+    }
+
+    @Then("{fieldVar} contains strings of length outside {int} and {int}")
```

Comment:
another `inclusively` description (code uses `(minInclusive, maxInclusive)` )

Reply:
No, agreed to treat this as a full complement of between (which is inclusive) therefore this one should be exclusive